x-pack/metricbeat/module/aws/cloudwatch/metadata/ec2/ec2.go

```diff
@@ -55,7 +55,7 @@ func AddMetadata(regionName string, awsConfig awssdk.Config, fips_enabled bool,
 			if output.InstanceType != "" {
 				_, _ = events[eventIdentifier].RootFields.Put("cloud.machine.type", output.InstanceType)
 			} else {
-				logp.Error(fmt.Errorf("InstanceType is empty"))
+				logp.L().Named("cloudwatch").Error("InstanceType is empty")
```

Comment:
What do you think about passing the logger to the function? This would avoid creating additional loggers in case of errors.

Reply:
Makes sense, this was one of the original ideas that we've discussed. 

It might not be the cleanest, but at least we keep the flow of execution intact while also properly logging the error.